src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -305,43 +321,219 @@ function compute_gwd_fields(domain) result(iErr)
+   !    list of tiles tilesHead, box coordinates box_x, box_y, and path to 
+   !    static dataset. The function first searches the linked list to locate
+   !    the tile, and if search fails, adds the target tile to the linked list
+   !    after reading in the data for the tile from disk 
+   !-----------------------------------------------------------------------
+   function get_tile_from_box_point(tilesHead, box_x, box_y, path, sub_path) result(thisTile)
+
+      implicit none
+
+      integer, intent(in) :: box_x, box_y
+      type(mpas_gwd_tile_type), pointer, intent(in) :: tilesHead
+      type(mpas_gwd_tile_type), pointer :: thisTile
+      character(len=*), intent(in) :: path
+      character(len=*), intent(in) :: sub_path
+
+      integer :: tile_start_x, tile_start_y, tile_start_x_topo
+      
+      ! Need special handling for the x-coordinates of topo tiles, due to the shift by topo_shift
+      ! in certain datasets. We use tile_start_x, tile_start_y to search for tiles and open landmask tiles,
+      ! whereas tile_start_x_topo is only required to open the correct topo tiles from disk
+      if (box_x > topo_shift) then
+         tile_start_x_topo = floor( real(box_x - topo_shift - 1) / real(tile_x)) * tile_x + 1
+      else
+         tile_start_x_topo = floor( real(box_x + topo_shift - 1) / real(tile_x)) * tile_x + 1
+      end if
+      tile_start_x = floor( real(box_x - 1) / real(tile_x)) * tile_x + 1
+      tile_start_y = floor( real(box_y - 1) / real(tile_y)) * tile_y + 1
+      
+      thisTile => tilesHead
+      ! loop over tiles
+      do while (associated(thisTile))
+
+         if (thisTile%tile_start_x==tile_start_x .and. thisTile%tile_start_y==tile_start_y) then 
+            exit
+         end if
+
+         thisTile => thisTile % next
+
+      end do   ! associated(thisTile)
+
+      ! Could not find such a tile, so we add the tile to the front of the linked list
+      if (.not. associated(thisTile)) then 
+         thisTile => add_tile(tilesHead, tile_start_x, tile_start_y, tile_start_x_topo, path, sub_path)
+      end if
+
+   end function get_tile_from_box_point
+
+
+
+   !***********************************************************************
+   !
+   !  function add_tile
+   !
+   !> \brief   Routine to read in a new topo and landmask tile
+   !> \author  Abishek Gopal
+   !> \date    05 Sep 2024
+   !> \details 
+   !>  Routine to read in a new topo and landmask tile, given the tile 
+   !    coordinates
+   !-----------------------------------------------------------------------
+   function add_tile(tilesHead, tile_start_x, tile_start_y, tile_start_x_topo, path, sub_path) result(newTile)
+
+      implicit none
+
+      integer :: iErr
+      integer, intent(in) :: tile_start_x, tile_start_y, tile_start_x_topo
+      type(mpas_gwd_tile_type), pointer :: tilesHead
+      character(len=*), intent(in) :: path
+      character(len=*), intent(in) :: sub_path
+      type(mpas_gwd_tile_type), pointer :: newTile
```

Comment:
I think there's a general preference throughout the MPAS code for declaring in the following order:
1. Dummy arguments (in the order they appear in the arg list, except where a dummy argument array is dimensioned by other arguments, in which case the dimensions must be declared first)
2. Return value
3. Local variables

See, e.g., this code: https://github.com/MPAS-Dev/MPAS-Model/blob/v8.2.2/src/framework/mpas_stream_inquiry.F#L105-L118

Reply:
Thanks! Changed the declaration order in this and other routines.